### ISSUE TITLE
NSQ Study Materialization Authority: environment, raw bytes, and observation identities

### ISSUE THREAD

**Issue by sidhulyalkar** [sidhulyalkar/neurOS-v1#93]
## Why

NSQ now has strong protocol, split, preprocessing, model-state, scorecard, and failure-preserving authority. Kumar2024 also binds the exact processed arrays used for qualification. Two reviewer-facing gaps remain before any promoted neural efficacy run:

1. the execution environment is represented by a hand-picked package subset rather than a canonical realized environment identity;
2. dataset lineage is intentionally `partial` because the actual downloaded raw materialization is not byte-addressed;
3. case membership is machine-auditable through integer row-index hashes but not yet human-auditable as participant/session/run/local-epoch identities.

These should be solved once as a reusable NSQ primitive, not as Kumar2024-only glue.

## North-star invariant

> A promoted NSQ result must be traceable from raw study materialization through processed observation identities and exact execution environment to the qualified model state and score, without relying on machine-local paths or undocumented reconstruction steps.

This issue is a prerequisite for promoted Kumar2024 EEGNet/all-subject efficacy execution under #90 and #27. It is separate from runtime transform descriptor lineage in #74.

## Proposed architecture

Introduce a content-addressed `StudyMaterializationAuthority` (name can change if a better neutral name emerges) composed from three independently inspectable authorities.

### 1. Environment authority

Canonical manifest should record at least:

- Python implementation and exact version;
- OS/platform family and architecture;
- every installed Python distribution relevant to the study environment as sorted `{normalized_name, version}` entries;
- NumPy/SciPy/sklearn/MNE/MOABB/pyRiemann/Braindecode/skorch/Torch versions when present;
- accelerator/device type;
- CUDA/cuDNN/runtime versions when applicable;
- deterministic-algorithm flags relevant to the method;
- neurOS workspace package versions and, when running from source, repository commit identity.

Requirements:

- canonical JSON independent of installation order;
- machine-local absolute paths excluded from scientific identity;
- environment identity must change when a scientifically relevant dependency/version changes;
- volatile host noise such as hostname, PID, timestamps, temp paths, runner IDs, and free memory must not enter the identity root;
- telemetry may record volatile fields separately, but they cannot masquerade as reproducibility authority.

### 2. Raw materialization authority

For the exact dataset files actually consumed by the loader, persist a manifest containing:

```text
logical dataset id
upstream dataset/version/DOI identity
canonical logical file name or dataset-relative path
byte size
SHA-256
```

Requirements:

- hash raw file bytes, not the cache directory name;
- strip machine-specific root paths;
- sort deterministically;
- distinguish source archive/container identity from extracted member identity when both matter;
- do not copy or redistribute licensed raw data into the evidence bundle; publish hashes/metadata only;
- fail closed for promoted evidence if the loader cannot explain which materialized files produced the processed observations;
- allow an explicit weaker lineage tier for exploratory integrations, but never silently upgrade it to complete provenance.

For Kumar2024, bind this authority to the exact MOABB materialization used for subjects/sessions in the study.

### 3. Observation identity authority

Every processed row should have a stable, human-auditable identity derived from available upstream metadata, e.g.:

```text
row_index
participant
session
run
local_epoch / trial ordinal
upstream event identity when available
```

The identity manifest must preserve the exact processed row order while also exposing a SHA for each role set used by NSQ:

- source history;
- labeled calibration;
- internal method validation where applicable;
- final assessment.

Requirements:

- row reorder changes the processed observation authority;
- duplicate/ambiguous identities fail closed;
- integer NumPy indices remain available for execution but are not the only reviewer interface;
- method code receives only observations authorized by NSQ. Human-readable identities are evidence metadata, not a side channel for final-assessment labels.

### 4. Composed study materialization root

Compose:

```text
raw_materialization_sha256
        +
processed/preprocessing_authority_sha256
        +
observation_identity_sha256
        +
environment_authority_sha256
        -> study_materialization_sha256
```

Bind this root into the NSQ execution context / study manifest so downstream case results and model-state evidence can prove which materialized study they belong to.

Do not overload `DatasetLineage` with runtime environment semantics if a separate typed composition is clearer. The important property is a typed, inspectable, content-addressed chain.

## Adversarial tests

- same files under a different local cache root -> same raw materialization SHA;
- one raw byte changes -> different SHA;
- missing/extra consumed file -> different or rejected authority;
- package installation order changes -> same environment SHA;
- relevant package version changes -> different environment SHA;
- hostname/timestamp/temp path changes -> same environment SHA;
- row reorder -> different observation identity / processed authority;
- duplicate participant/session/run/epoch identity -> fail closed;
- case role membership rendered from human identities agrees exactly with existing integer-index hashes;
- model adapter cannot access final-assessment observations through the identity surface;
- evidence bundle verification detects tampering without rerunning training.

## Kumar2024 acceptance gates

- [ ] exact consumed raw files are enumerated and byte-hashed;
- [ ] machine-local cache roots do not affect identity;
- [ ] exact realized Python dependency environment is content-addressed;
- [ ] source commit/workspace identity is recorded for source executions;
- [ ] every processed trial exposes participant/session/run/local-epoch identity;
- [ ] source/calibration/validation/final role sets are human-auditable and hash-bound;
- [ ] study manifest binds the composed materialization root;
- [ ] `DatasetLineage` completeness is upgraded only if its stated ancestry contract is genuinely satisfied;
- [ ] no promoted EEGNet final-assessment result is generated while this authority is still being tuned.

## Scope boundary

This is evidence/reproducibility infrastructure. It does **not** prove hardware validity, physiological signal quality, participant representativeness, causal efficacy, or clinical benefit.

It also does not replace #74. Runtime descriptor lineage answers whether transformed live data still satisfy a decoder input contract; study materialization authority answers exactly what offline study bytes/environment/observations produced a scientific result.
